src/_pytest/mark/__init__.py

```diff
@@ -172,6 +172,9 @@ def from_item(cls, item: "Item") -> "KeywordMatcher":
         # Add the markers to the keywords as we no longer handle them correctly.
         mapped_names.update(mark.name for mark in item.iter_markers())
 
+        # Remove names that are private(characterised by starting with '_')
+        mapped_names = {name for name in mapped_names if not name.startswith("_")}
```

Comment:
The exclusion should be done earlier, in

https://github.com/pytest-dev/pytest/blob/364bbe42fc02d435e6d94d13ad3accc7e00875fd/src/_pytest/python.py#L1593

rather than here.

Reply:
@bluetech, thanks for the suggetion.
Upon looking at `self.keywords` which's updated in the referenced line, it seems that this is associated with the options passed to the pytest command rather than the list of the nodes that are found. So I am not sure if the action of removing the private nodes from the list would be possible here.

Please let me know if I am missing something.

Reply:
You are right, I was confusing things (or rather, pytest itself is confusing things :grinning:). Your change is done in the right place, although, it probably should be done in *both* cases, as they are mostly supposed to be consistent.